tests/ranged_balance.cpp

```diff
@@ -334,6 +338,27 @@ static void range_test( const std::array<double, 5> &test_thresholds )
             WARN( "No matching dispersion found" );
         } else {
             WARN( "Range: " << r << " Dispersion: " << found_dispersion );
+            data.push_back( found_dispersion );
+        }
+    }
+    if( write_data )
+    {
+        if( std::equal( data.begin(), data.end(),
+                        Creature::dispersion_for_even_chance_of_good_hit.begin(),
+                        Creature::dispersion_for_even_chance_of_good_hit.end(),
+                        []( const int a, const int b ) -> bool {
+            return a > 0 && b > 0 && std::abs( static_cast<float>( a - b ) / b ) < 0.05;
+        } ) )
```

Comment:
Under this logic it only writes to the JSON file if the results *hasn't* changed significantly. What I meant was using the unit test's `REQUIRE()` macro to check the dispersion values, and fail the test if they changed considerably, so if the values are changed in the future, we can be automatically notified of the change.

Reply:
Should it use REQUIRE_FALSE?

Reply:
If you require `std::equal(...)` to be false, you'll get a test failure when the values have *not* changed. I don't think that's what we want here.

Reply:
That makes sense. I've noticed differences of more than 10% after running the test several times in a row. Would it be better to raise the limit to something like 15% or to accept some false-positives?

Reply:
That makes sense. I've noticed differences of more than 10% after running the test several times in a row. Would it be better to raise the limit to something like 15% or to accept some false-positives?

Reply:
Bump it to 15% then. Sounds like 5% would cause too many false positives.